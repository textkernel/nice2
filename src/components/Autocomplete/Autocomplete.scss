.Autocomplete {
    position: relative;
    box-shadow: 0 2px 2px 0 rgba(0, 0, 0, 0.1);
    background-color: var(--color-background);
    transition: all var(--transition-duration);
    border-radius: var(--border-radius);

    &--focused {
        position: absolute;
    }

    &--isProminent {
        transform-origin: left top;

        &.Autocomplete--focused {
            transform: scale(1.1);
            box-shadow: 0 5px 15px 0 rgba(0, 0, 0, 0.4);
        }
    }

    &__wrapper {
        cursor: pointer;
        display: flex;
        flex-wrap: wrap;
        align-items: baseline;

        &--focused {
            margin-bottom: var(--spacing-normal);
            border-bottom: 1px solid var(--border-color-light);
        }
    }

    &__spacedElem {
        margin: var(--spacing-normal);
    }

    &__input {
        border: 0;
        margin: var(--spacing-normal);
        flex: 1;
        font-size: var(--font-size-normal);
        min-width: 150px;

        &:focus {
            outline: none;
        }
    }

    &__list--focused {
        max-height: 200px;
        overflow: auto;
    }

    &__loaderContainer {
        width: 250px;
    }
}
