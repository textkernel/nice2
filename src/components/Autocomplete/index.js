export { default as Autocomplete } from './Autocomplete';
export { default as ItemTag } from './ItemTag';
