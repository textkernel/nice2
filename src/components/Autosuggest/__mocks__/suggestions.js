export const SUGGESTIONS = [
    { name: 'Sun' },
    { name: 'Moon' },
    { name: 'Mercury' },
    { name: 'Venus' },
    { name: 'Earth' },
    { name: 'Mars' },
    { name: 'Jupiter' },
    { name: 'Saturn' },
    { name: 'Neptun' }
];

export const SUGGESTION_TO_STRING = item => (item ? item.name : '');
