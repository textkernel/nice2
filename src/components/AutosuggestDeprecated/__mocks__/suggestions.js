export const SUGGESTIONS = [
    { name: 'Sun' },
    { name: 'Moon' },
    { name: 'Mercury' },
    { name: 'Venus' },
    { name: 'Earth' },
    { name: 'Mars' },
    { name: 'Jupiter' },
    { name: 'Saturn' },
    { name: 'Neptun' },
];

export const COMPLEX_SUGGESTIONS = [
    { name: 'Sun', type: 'Star', count: 1234 },
    { name: 'Moon', type: 'Star', count: 1234 },
    { name: 'Mercury', type: 'Planet', count: 1234 },
    { name: 'Venus', type: 'Planet', count: 1234 },
    { name: 'Earth', type: 'Planet', count: 1234 },
    { name: 'Mars', type: 'Planet', count: 1234 },
    { name: 'Jupiter', type: 'Planet', count: 1234 },
    { name: 'Saturn', type: 'Planet', count: 1234 },
    { name: 'Neptun', type: 'Planet', count: 1234 },
];

export const SUGGESTION_TO_STRING = (item) => (item ? item.name : '');
