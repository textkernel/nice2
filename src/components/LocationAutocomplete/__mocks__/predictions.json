[
    {
        "description": "Truro, UK",
        "id": "51b6747d060368917397ce3cc25abe2089302262",
        "matched_substrings": [
            {
                "length": 2,
                "offset": 0
            }
        ],
        "place_id": "ChIJdRpa1XwQa0gRtAcdle9HY2E",
        "reference": "ChIJdRpa1XwQa0gRtAcdle9HY2E",
        "structured_formatting": {
            "main_text": "Truro",
            "main_text_matched_substrings": [
                {
                    "length": 2,
                    "offset": 0
                }
            ],
            "secondary_text": "UK"
        },
        "terms": [
            {
                "offset": 0,
                "value": "Truro"
            },
            {
                "offset": 7,
                "value": "UK"
            }
        ],
        "types": ["locality", "political", "geocode"]
    },
    {
        "description": "Trowbridge, UK",
        "id": "b10d478a406230745e98edd2d2a01f59097d5a59",
        "matched_substrings": [
            {
                "length": 2,
                "offset": 0
            }
        ],
        "place_id": "ChIJM6W0Ewp9cUgRLXIknC4a-Pw",
        "reference": "ChIJM6W0Ewp9cUgRLXIknC4a-Pw",
        "structured_formatting": {
            "main_text": "Trowbridge",
            "main_text_matched_substrings": [
                {
                    "length": 2,
                    "offset": 0
                }
            ],
            "secondary_text": "UK"
        },
        "terms": [
            {
                "offset": 0,
                "value": "Trowbridge"
            },
            {
                "offset": 12,
                "value": "UK"
            }
        ],
        "types": ["locality", "political", "geocode"]
    },
    {
        "description": "Tring, UK",
        "id": "8b1f6899d3555c6aae3a2d383ed96c7649f3ccf8",
        "matched_substrings": [
            {
                "length": 2,
                "offset": 0
            }
        ],
        "place_id": "ChIJ33SmFitYdkgR3-ErzH_p6Bw",
        "reference": "ChIJ33SmFitYdkgR3-ErzH_p6Bw",
        "structured_formatting": {
            "main_text": "Tring",
            "main_text_matched_substrings": [
                {
                    "length": 2,
                    "offset": 0
                }
            ],
            "secondary_text": "UK"
        },
        "terms": [
            {
                "offset": 0,
                "value": "Tring"
            },
            {
                "offset": 7,
                "value": "UK"
            }
        ],
        "types": ["locality", "political", "geocode"]
    },
    {
        "description": "Troon, UK",
        "id": "c563f4c69998a376dac8e3b4122067d782e752f7",
        "matched_substrings": [
            {
                "length": 2,
                "offset": 0
            }
        ],
        "place_id": "ChIJxy8qU3jRiUgR-sKnFLHxiSM",
        "reference": "ChIJxy8qU3jRiUgR-sKnFLHxiSM",
        "structured_formatting": {
            "main_text": "Troon",
            "main_text_matched_substrings": [
                {
                    "length": 2,
                    "offset": 0
                }
            ],
            "secondary_text": "UK"
        },
        "terms": [
            {
                "offset": 0,
                "value": "Troon"
            },
            {
                "offset": 7,
                "value": "UK"
            }
        ],
        "types": ["locality", "political", "geocode"]
    },
    {
        "description": "Tredegar, UK",
        "id": "4dca94194f54dee0879b9d368379eb5d09ec838d",
        "matched_substrings": [
            {
                "length": 2,
                "offset": 0
            }
        ],
        "place_id": "ChIJdQjXYyYkbkgRTRfZ8WAt6lI",
        "reference": "ChIJdQjXYyYkbkgRTRfZ8WAt6lI",
        "structured_formatting": {
            "main_text": "Tredegar",
            "main_text_matched_substrings": [
                {
                    "length": 2,
                    "offset": 0
                }
            ],
            "secondary_text": "UK"
        },
        "terms": [
            {
                "offset": 0,
                "value": "Tredegar"
            },
            {
                "offset": 10,
                "value": "UK"
            }
        ],
        "types": ["locality", "political", "geocode"]
    }
]
