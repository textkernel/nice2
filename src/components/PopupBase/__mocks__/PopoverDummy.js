import * as React from 'react';
import PropTypes from 'prop-types';
import { Button } from '../../..';

const PopoverDummy = React.forwardRef((props, ref) => {
    const { setPopupVisibility } = props;
    const popupStyle = {
        backgroundColor: '#FFEEDD',
        padding: '10px',
        borderRadius: '5px',
        border: '1px solid #CCC',
        boxShadow: '0 3px 6px 6px rgba(0, 0, 0, 0.05)',
    };
    return (
        <div style={popupStyle} ref={ref}>
            <p>I am the content of pop up.</p>
            <p>Can be a form or anything.</p>
            <Button onClick={() => setPopupVisibility(false)}>Close me!</Button>
        </div>
    );
});

PopoverDummy.propTypes = {
    setPopupVisibility: PropTypes.func.isRequired,
};

PopoverDummy.displayName = 'Popover';

export default PopoverDummy;
