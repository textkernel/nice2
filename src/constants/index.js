export const INPUT_TYPES = [
    'text',
    'password',
    'email',
    'date',
    'datetime-local',
    'month',
    'tel',
    'number',
    'url',
    'search',
    'time',
    'week',
];

export const LIST_NAVIGATION_DIRECTIONS = {
    UP: 'ArrowUp',
    DOWN: 'ArrowDown',
};

export const CROSS_CHAR = 'Ã—';

export * from './style-related';
export * from './autocomplete';
export * from './keyboard';
