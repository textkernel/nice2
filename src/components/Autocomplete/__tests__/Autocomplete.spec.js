/* eslint-disable */
import React from 'react';
import toJson from 'enzyme-to-json';
import Autocomplete from '../Autocomplete';
import { SUGGESTIONS, SUGGESTION_TO_STRING } from '../__mocks__/suggestions';
import { NUMBER_OF_SUGGESTION_LOADING_PLACEHOLDERS } from '../../../constants';

describe('Autocomplete', () => {
    const suggestionToString = SUGGESTION_TO_STRING;
    let selectedPlaceholder = '';
    const inputPlaceholder = 'type here...';
    const noSuggestionsPlaceholder = 'No suggestions...';
    const clearTitle = 'Clear';
    const mockOnSelectionChange = jest.fn();
    const mockOnInputValueChange = jest.fn();
    const mockonClearAllSelected = jest.fn();
    const mockOnBlur = jest.fn();

    let suggestionsList = null;
    const getSuggestions = () => suggestionsList;
    let selectedSuggestions = [];

    let wrapper;
    let instance;

    const setFocusOnInput = () => wrapper.find('.Autocomplete__input').simulate('click');

    beforeEach(() => {
        wrapper = mount(
            <Autocomplete
                selectedSuggestions={selectedSuggestions}
                getSuggestions={getSuggestions}
                suggestionToString={suggestionToString}
                selectedPlaceholder={selectedPlaceholder}
                inputPlaceholder={inputPlaceholder}
                clearTitle={clearTitle}
                noSuggestionsPlaceholder={noSuggestionsPlaceholder}
                onSelectionChange={mockOnSelectionChange}
                onInputValueChange={mockOnInputValueChange}
                onBlur={mockOnBlur}
                onClearAllSelected={mockonClearAllSelected}
                showClearButton
            />
        );
        instance = wrapper.find('Autocomplete').instance();
    });

    afterEach(() => {
        suggestionsList = null;
        selectedSuggestions = [];
    });

    describe('rendering', () => {
        it('should initally render empty component correctly', () => {
            expect(toJson(wrapper)).toMatchSnapshot();
            expect(wrapper.state('focused')).toBeFalsy();
        });
        it('should render empty component correctly when focused', () => {
            setFocusOnInput();
            expect(wrapper.find('li')).toHaveLength(0);
            expect(wrapper.state('focused')).toBeTruthy();
        });
        it('should render noSuggestions placeholder when empty suggestions list is passed', () => {
            suggestionsList = [];
            wrapper.find('input').simulate('change', { target: { value: 'driver' } });
            setFocusOnInput();

            expect(toJson(wrapper)).toMatchSnapshot();
            expect(wrapper.find('li')).toHaveLength(1);
            expect(
                wrapper
                    .find('li')
                    .childAt(0)
                    .text()
            ).toEqual(noSuggestionsPlaceholder);
        });
        it('should render all suggestions from the list', () => {
            suggestionsList = SUGGESTIONS.slice(0, 4);
            setFocusOnInput();

            expect(toJson(wrapper)).toMatchSnapshot();
            expect(wrapper.find('li')).toHaveLength(suggestionsList.length);
        });
        it('should render selection placeholder when component is not focused', () => {
            selectedPlaceholder = 'Current selection';
            wrapper.setProps({ selectedPlaceholder });
            expect(wrapper.find('.Autocomplete__spacedElem').text()).toEqual(selectedPlaceholder);
        });
        it('should render tag for each selected selection when component is focused', () => {
            selectedSuggestions = SUGGESTIONS.slice(0, 5);
            wrapper.setProps({ selectedSuggestions });
            setFocusOnInput();

            expect(wrapper.find('ItemTag')).toHaveLength(selectedSuggestions.length);
        });
        it('should render clear button only if there are selected suggestion', () => {
            expect(wrapper.find('.InputWrapper__clearButton')).toHaveLength(0);

            selectedSuggestions = SUGGESTIONS.slice(0, 5);
            wrapper.setProps({ selectedSuggestions });

            expect(wrapper.find('.InputWrapper__clearButton')).toHaveLength(2);
        });
        it('should not render clear button if component is in focus', () => {
            selectedSuggestions = SUGGESTIONS.slice(0, 5);
            wrapper.setProps({ selectedSuggestions });

            expect(wrapper.find('.InputWrapper__clearButton')).toHaveLength(2);
            setFocusOnInput();

            expect(wrapper.find('.InputWrapper__clearButton')).toHaveLength(0);
        });

        it(`should render ${NUMBER_OF_SUGGESTION_LOADING_PLACEHOLDERS} loaders if isLoading is true`, () => {
            wrapper.find('input').simulate('change', { target: { value: 'driver' } });
            wrapper.setProps({ isLoading: true });

            expect(wrapper.find('li')).toHaveLength(NUMBER_OF_SUGGESTION_LOADING_PLACEHOLDERS);
        });
    });
    describe('focusing and blurring the search field', () => {
        it('should set focus when wrapper element is clicked', () => {
            const inputEl = instance.inputRef.current;
            const focusSpy = jest.spyOn(inputEl, 'focus');

            expect(wrapper.state('focused')).toBeFalsy();
            expect(focusSpy).not.toHaveBeenCalled();

            wrapper.find('.Autocomplete__wrapper').simulate('click');

            expect(wrapper.state('focused')).toBeTruthy();
            expect(focusSpy).toHaveBeenCalledTimes(1);
        });
        it('should blur on pressing Escape button', () => {
            const inputEl = instance.inputRef.current;
            const blurSpy = jest.spyOn(inputEl, 'blur');

            setFocusOnInput();

            expect(wrapper.state('focused')).toBeTruthy();

            wrapper.find('input').simulate('keyDown', { key: 'Escape' });

            expect(wrapper.state('focused')).toBeFalsy();
            expect(blurSpy).toHaveBeenCalled();
            expect(mockOnBlur).toHaveBeenCalled();
        });
        it('should blur on pressing Tab button', () => {
            const inputEl = instance.inputRef.current;
            const blurSpy = jest.spyOn(inputEl, 'blur');

            setFocusOnInput();

            expect(wrapper.state('focused')).toBeTruthy();

            wrapper.find('input').simulate('keyDown', { key: 'Tab' });

            /**
             * This line makes test fail.
             * According to that the component doesn't lose the 'focused' property, but in fact it does.
             * The reason is JSDom rendered here has no other tabular element.
             * expect(wrapper.state('focused')).toBeFalsy();
             */
            expect(blurSpy).toHaveBeenCalled();
            expect(mockOnBlur).toHaveBeenCalled();
        });
        describe('in single select mode', () => {
            it('should not stay focused when suggestion is selected', () => {
                suggestionsList = SUGGESTIONS.slice(1, 20);

                expect(wrapper.prop('isMultiselect')).toBeFalsy();
                expect(wrapper.state('focused')).toBeFalsy();

                setFocusOnInput();
                expect(wrapper.state('focused')).toBeTruthy();

                wrapper
                    .find('li')
                    .at(0)
                    .childAt(0)
                    .simulate('click');

                expect(wrapper.state('focused')).toBeFalsy();
                expect(wrapper.find('li')).toHaveLength(0);
            });
        });
        describe('in multiselect mode', () => {
            it('should stay focused when suggestion is selected', () => {
                suggestionsList = SUGGESTIONS.slice(1, 20);
                wrapper.setProps({ isMultiselect: true });

                // expect(wrapper.prop('isMultiselect')).toBeFalsy();
                expect(wrapper.state('focused')).toBeFalsy();

                setFocusOnInput();
                expect(wrapper.state('focused')).toBeTruthy();

                wrapper
                    .find('li')
                    .at(0)
                    .childAt(0)
                    .simulate('click');

                expect(wrapper.state('focused')).toBeTruthy();
                expect(wrapper.find('li')).not.toHaveLength(0);
            });
            it('should stay focused when suggestion is deleted from selected list', () => {
                suggestionsList = SUGGESTIONS.slice(0, 20);
                selectedSuggestions = SUGGESTIONS.slice(1, 4);
                wrapper.setProps({ selectedSuggestions });

                setFocusOnInput();
                expect(wrapper.state('focused')).toBeTruthy();

                wrapper.find('input').simulate('keyDown', { key: 'Backspace' });

                expect(wrapper.state('focused')).toBeTruthy();
            });
        });
    });
    describe('callbacks', () => {
        describe('onSelectionChange', () => {
            it('should be called on clicking on a suggestion', () => {
                suggestionsList = SUGGESTIONS.slice(1, 20);
                setFocusOnInput();

                expect(mockOnSelectionChange).not.toHaveBeenCalled();

                wrapper
                    .find('li')
                    .first()
                    .children()
                    .simulate('click');

                expect(mockOnSelectionChange).toHaveBeenCalled();
            });
            it('should be called on deleting a suggestion by clicking on the x button next to it', () => {
                selectedSuggestions = SUGGESTIONS.slice(0, 5);
                wrapper.setProps({ selectedSuggestions });
                setFocusOnInput();

                expect(mockOnSelectionChange).not.toHaveBeenCalled();
                expect(wrapper.find('ItemTag')).toHaveLength(selectedSuggestions.length);

                wrapper
                    .find('ItemTag')
                    .at(2)
                    .find('button')
                    .simulate('click');

                expect(mockOnSelectionChange).toHaveBeenCalledWith(selectedSuggestions[2]);
            });
            it('should be called on deleting a suggestion by hitting backspace in the empty input field', () => {
                selectedSuggestions = SUGGESTIONS.slice(0, 5);
                wrapper.setProps({ selectedSuggestions });
                setFocusOnInput();

                expect(mockOnSelectionChange).not.toHaveBeenCalled();

                wrapper.find('input').simulate('keyDown', { key: 'Backspace' });

                expect(mockOnSelectionChange).toHaveBeenCalled();
            });
            it('should not be called on deleting a suggestion by hitting backspace in an input field with value', () => {
                selectedSuggestions = SUGGESTIONS.slice(0, 5);
                wrapper.setProps({ selectedSuggestions });
                setFocusOnInput();
                wrapper.find('input').simulate('change', { target: { value: 'driver' } });

                expect(mockOnSelectionChange).not.toHaveBeenCalled();

                wrapper.find('input').simulate('keyDown', { key: 'Backspace' });

                expect(mockOnSelectionChange).not.toHaveBeenCalled();
            });
        });
        it('should call onInputValueChange when typing into input field', () => {
            expect(mockOnInputValueChange).not.toHaveBeenCalled();

            wrapper.find('input').simulate('change', { target: { value: 'driver' } });

            expect(mockOnInputValueChange).toHaveBeenCalled();
        });
        it('should call onClearAllSelected when clear button was clicked', () => {
            selectedSuggestions = SUGGESTIONS.slice(0, 5);
            wrapper.setProps({ selectedSuggestions });
            expect(mockonClearAllSelected).not.toHaveBeenCalled();

            wrapper
                .find('.InputWrapper__clearButton')
                .at(1)
                .simulate('click');

            expect(mockonClearAllSelected).toHaveBeenCalled();
        });
        it('should clear the input field when a suggestion was selected', () => {
            const textInputValue = 'driver';
            suggestionsList = SUGGESTIONS.slice(1, 20);
            wrapper.find('input').simulate('change', { target: { value: textInputValue } });

            expect(wrapper.find('input').props().value).toEqual(textInputValue);

            wrapper
                .find('li')
                .first()
                .children()
                .simulate('click');

            expect(wrapper.find('input').props().value).toEqual('');
        });
    });
});
