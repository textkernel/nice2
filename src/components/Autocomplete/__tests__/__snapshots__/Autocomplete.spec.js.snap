// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Autocomplete rendering should initally render empty component correctly 1`] = `
<Autocomplete
  clearTitle="Clear"
  getSuggestions={[Function]}
  iconNode={null}
  inputPlaceholder="type here..."
  isLoading={false}
  isMultiselect={false}
  noSuggestionsPlaceholder="No suggestions..."
  onBlur={[MockFunction]}
  onClearAllSelected={[MockFunction]}
  onInputValueChange={[MockFunction]}
  onSelectionChange={[MockFunction]}
  selectedPlaceholder=""
  selectedSuggestions={Array []}
  showClearButton={true}
  suggestionToString={[Function]}
>
  <Downshift
    defaultHighlightedIndex={0}
    defaultIsOpen={false}
    environment={[Window]}
    getA11yStatusMessage={[Function]}
    inputValue=""
    itemToString={[Function]}
    onChange={[Function]}
    onInputValueChange={[Function]}
    onOuterClick={[Function]}
    onSelect={[Function]}
    onStateChange={[Function]}
    onUserAction={[Function]}
    scrollIntoView={[Function]}
    selectedItemChanged={[Function]}
    stateReducer={[Function]}
    suppressRefError={false}
  >
    <InputWrapper
      aria-expanded={false}
      aria-haspopup="listbox"
      aria-labelledby="downshift-0-label"
      aria-owns={null}
      className="Autocomplete"
      clearLabel="Clear"
      onClear={[Function]}
      role="combobox"
      showClearButton={false}
    >
      <div
        aria-expanded={false}
        aria-haspopup="listbox"
        aria-labelledby="downshift-0-label"
        aria-owns={null}
        className="InputWrapper Autocomplete"
        role="combobox"
      >
        <div
          className="Autocomplete__wrapper"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="searchbox"
          tabIndex="0"
        >
          <input
            aria-activedescendant={null}
            aria-autocomplete="list"
            aria-controls={null}
            aria-labelledby="downshift-0-label"
            autoComplete="off"
            className="Autocomplete__input"
            id="downshift-0-input"
            onBlur={[Function]}
            onChange={[Function]}
            onKeyDown={[Function]}
            placeholder="type here..."
            value=""
          />
        </div>
        <List
          aria-labelledby="downshift-0-label"
          id="downshift-0-menu"
          isDivided={false}
          role="listbox"
        >
          <ul
            aria-labelledby="downshift-0-label"
            className="List"
            id="downshift-0-menu"
            role="listbox"
          />
        </List>
      </div>
    </InputWrapper>
  </Downshift>
</Autocomplete>
`;

exports[`Autocomplete rendering should render all suggestions from the list 1`] = `
<Autocomplete
  clearTitle="Clear"
  getSuggestions={[Function]}
  iconNode={null}
  inputPlaceholder="type here..."
  isLoading={false}
  isMultiselect={false}
  noSuggestionsPlaceholder="No suggestions..."
  onBlur={[MockFunction]}
  onClearAllSelected={[MockFunction]}
  onInputValueChange={[MockFunction]}
  onSelectionChange={[MockFunction]}
  selectedPlaceholder=""
  selectedSuggestions={Array []}
  showClearButton={true}
  suggestionToString={[Function]}
>
  <Downshift
    defaultHighlightedIndex={0}
    defaultIsOpen={false}
    environment={[Window]}
    getA11yStatusMessage={[Function]}
    inputValue=""
    itemToString={[Function]}
    onChange={[Function]}
    onInputValueChange={[Function]}
    onOuterClick={[Function]}
    onSelect={[Function]}
    onStateChange={[Function]}
    onUserAction={[Function]}
    scrollIntoView={[Function]}
    selectedItemChanged={[Function]}
    stateReducer={[Function]}
    suppressRefError={false}
  >
    <InputWrapper
      aria-expanded={true}
      aria-haspopup="listbox"
      aria-labelledby="downshift-3-label"
      aria-owns="downshift-3-menu"
      className="Autocomplete Autocomplete--isOpen"
      clearLabel="Clear"
      onClear={[Function]}
      role="combobox"
      showClearButton={false}
    >
      <div
        aria-expanded={true}
        aria-haspopup="listbox"
        aria-labelledby="downshift-3-label"
        aria-owns="downshift-3-menu"
        className="InputWrapper Autocomplete Autocomplete--isOpen"
        role="combobox"
      >
        <div
          className="Autocomplete__wrapper Autocomplete__wrapper--isOpen"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="searchbox"
          tabIndex="0"
        >
          <input
            aria-activedescendant="downshift-3-item-0"
            aria-autocomplete="list"
            aria-controls="downshift-3-menu"
            aria-labelledby="downshift-3-label"
            autoComplete="off"
            className="Autocomplete__input"
            id="downshift-3-input"
            onBlur={[Function]}
            onChange={[Function]}
            onKeyDown={[Function]}
            placeholder="type here..."
            value=""
          />
        </div>
        <List
          aria-labelledby="downshift-3-label"
          className="Autocomplete__list--isOpen"
          id="downshift-3-menu"
          isDivided={false}
          role="listbox"
        >
          <ul
            aria-labelledby="downshift-3-label"
            className="List Autocomplete__list--isOpen"
            id="downshift-3-menu"
            role="listbox"
          >
            <ListItem
              aria-selected={true}
              className="List__item"
              disabled={false}
              highlightContext="brand"
              id="downshift-3-item-0"
              isSelected={true}
              key=".$Sun"
              onClick={[Function]}
              onMouseDown={[Function]}
              onMouseMove={[Function]}
              role="option"
            >
              <li
                aria-selected={true}
                className="ListItem ListItem--isSelected ListItem--highlightContext_brand ListItem--clickable List__item"
                id="downshift-3-item-0"
                onMouseDown={[Function]}
                onMouseMove={[Function]}
                role="option"
              >
                <div
                  className="ListItem__container"
                  onClick={[Function]}
                  role="presentation"
                >
                  <Text
                    context="default"
                    inline={true}
                    key=".0"
                    size="normal"
                  >
                    <span>
                      Sun
                    </span>
                  </Text>
                </div>
              </li>
            </ListItem>
            <ListItem
              aria-selected={false}
              className="List__item"
              disabled={false}
              highlightContext="brand"
              id="downshift-3-item-1"
              isSelected={false}
              key=".$Moon"
              onClick={[Function]}
              onMouseDown={[Function]}
              onMouseMove={[Function]}
              role="option"
            >
              <li
                aria-selected={false}
                className="ListItem ListItem--highlightContext_brand ListItem--clickable List__item"
                id="downshift-3-item-1"
                onMouseDown={[Function]}
                onMouseMove={[Function]}
                role="option"
              >
                <div
                  className="ListItem__container"
                  onClick={[Function]}
                  role="presentation"
                >
                  <Text
                    context="default"
                    inline={true}
                    key=".0"
                    size="normal"
                  >
                    <span>
                      Moon
                    </span>
                  </Text>
                </div>
              </li>
            </ListItem>
            <ListItem
              aria-selected={false}
              className="List__item"
              disabled={false}
              highlightContext="brand"
              id="downshift-3-item-2"
              isSelected={false}
              key=".$Mercury"
              onClick={[Function]}
              onMouseDown={[Function]}
              onMouseMove={[Function]}
              role="option"
            >
              <li
                aria-selected={false}
                className="ListItem ListItem--highlightContext_brand ListItem--clickable List__item"
                id="downshift-3-item-2"
                onMouseDown={[Function]}
                onMouseMove={[Function]}
                role="option"
              >
                <div
                  className="ListItem__container"
                  onClick={[Function]}
                  role="presentation"
                >
                  <Text
                    context="default"
                    inline={true}
                    key=".0"
                    size="normal"
                  >
                    <span>
                      Mercury
                    </span>
                  </Text>
                </div>
              </li>
            </ListItem>
            <ListItem
              aria-selected={false}
              className="List__item"
              disabled={false}
              highlightContext="brand"
              id="downshift-3-item-3"
              isSelected={false}
              key=".$Venus"
              onClick={[Function]}
              onMouseDown={[Function]}
              onMouseMove={[Function]}
              role="option"
            >
              <li
                aria-selected={false}
                className="ListItem ListItem--highlightContext_brand ListItem--clickable List__item"
                id="downshift-3-item-3"
                onMouseDown={[Function]}
                onMouseMove={[Function]}
                role="option"
              >
                <div
                  className="ListItem__container"
                  onClick={[Function]}
                  role="presentation"
                >
                  <Text
                    context="default"
                    inline={true}
                    key=".0"
                    size="normal"
                  >
                    <span>
                      Venus
                    </span>
                  </Text>
                </div>
              </li>
            </ListItem>
          </ul>
        </List>
      </div>
    </InputWrapper>
  </Downshift>
</Autocomplete>
`;

exports[`Autocomplete rendering should render noSuggestions placeholder when empty suggestions list is passed 1`] = `
<Autocomplete
  clearTitle="Clear"
  getSuggestions={[Function]}
  iconNode={null}
  inputPlaceholder="type here..."
  isLoading={false}
  isMultiselect={false}
  noSuggestionsPlaceholder="No suggestions..."
  onBlur={[MockFunction]}
  onClearAllSelected={[MockFunction]}
  onInputValueChange={
    [MockFunction] {
      "calls": Array [
        Array [
          "driver",
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  onSelectionChange={[MockFunction]}
  selectedPlaceholder=""
  selectedSuggestions={Array []}
  showClearButton={true}
  suggestionToString={[Function]}
>
  <Downshift
    defaultHighlightedIndex={0}
    defaultIsOpen={false}
    environment={[Window]}
    getA11yStatusMessage={[Function]}
    inputValue="driver"
    itemToString={[Function]}
    onChange={[Function]}
    onInputValueChange={[Function]}
    onOuterClick={[Function]}
    onSelect={[Function]}
    onStateChange={[Function]}
    onUserAction={[Function]}
    scrollIntoView={[Function]}
    selectedItemChanged={[Function]}
    stateReducer={[Function]}
    suppressRefError={false}
  >
    <InputWrapper
      aria-expanded={true}
      aria-haspopup="listbox"
      aria-labelledby="downshift-2-label"
      aria-owns="downshift-2-menu"
      className="Autocomplete Autocomplete--isOpen"
      clearLabel="Clear"
      onClear={[Function]}
      role="combobox"
      showClearButton={false}
    >
      <div
        aria-expanded={true}
        aria-haspopup="listbox"
        aria-labelledby="downshift-2-label"
        aria-owns="downshift-2-menu"
        className="InputWrapper Autocomplete Autocomplete--isOpen"
        role="combobox"
      >
        <div
          className="Autocomplete__wrapper Autocomplete__wrapper--isOpen"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="searchbox"
          tabIndex="0"
        >
          <input
            aria-activedescendant="downshift-2-item-0"
            aria-autocomplete="list"
            aria-controls="downshift-2-menu"
            aria-labelledby="downshift-2-label"
            autoComplete="off"
            className="Autocomplete__input"
            id="downshift-2-input"
            onBlur={[Function]}
            onChange={[Function]}
            onKeyDown={[Function]}
            placeholder="type here..."
            value="driver"
          />
        </div>
        <List
          aria-labelledby="downshift-2-label"
          className="Autocomplete__list--isOpen"
          id="downshift-2-menu"
          isDivided={false}
          role="listbox"
        >
          <ul
            aria-labelledby="downshift-2-label"
            className="List Autocomplete__list--isOpen"
            id="downshift-2-menu"
            role="listbox"
          >
            <ListItem
              className="List__item"
              disabled={true}
              highlightContext="default"
              isSelected={false}
              key=".0"
              onClick={null}
            >
              <li
                className="ListItem List__item"
              >
                <div
                  className="ListItem__container"
                  onClick={null}
                  role="presentation"
                >
                  <Text
                    context="muted"
                    inline={false}
                    key=".0"
                    size="normal"
                  >
                    <p
                      className="Text--context_muted"
                    >
                      No suggestions...
                    </p>
                  </Text>
                </div>
              </li>
            </ListItem>
          </ul>
        </List>
      </div>
    </InputWrapper>
  </Downshift>
</Autocomplete>
`;
